### PR TITLE
[Release] 2.15

### PR DESCRIPTION
**Pull request by abelgardep**
Implements #2813 
Needs https://github.com/owncloud/android-library/pull/319

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,33 @@
+## 2.15 (June 2020)
+- Login rearchitecture
+- Support for OpenId Connect
+- Biometrical lock
```
Comment (jesmrec) [inline CHANGELOG.md:4]: I'd specify the new implementation, so this is not a new feature. Something like `Biometrical lock using the Android 10 library` or similar.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+- UI improvements, including:
+  + New bottom navigation bar
+- Support for usernames with '+'
```
Comment (jesmrec) [inline CHANGELOG.md:7]: Indicate that this is available for server versions from 10.4.1

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+- Chunking adaption to oCIS
+- End of support for Android KitKat
```
Comment (jesmrec) [inline CHANGELOG.md:9]: I'd add the number version (4.4)

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/PublicShareDialogFragment.kt [hunk 4]
```diff
@@ -610,11 +610,9 @@ class PublicShareDialogFragment : DialogFragment() {
 
         override fun onCancelDatePicker() {
 
-            val expirationToggle = view!!.findViewById<SwitchCompat>(R.id.shareViaLinkExpirationSwitch)
-
             // If the date has not been set yet, uncheck the toggle
-            if (expirationToggle.isChecked && shareViaLinkExpirationValue?.text === "") {
-                expirationToggle.isChecked = false
+            if (shareViaLinkExpirationSwitch.isChecked && shareViaLinkExpirationSwitch?.text === "") {
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/PublicShareDialogFragment.kt:614]: `shareViaLinkExpirationSwitch?.text.isEmpty()` is not enough here?

File: owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/authentication/OCAuthenticationViewModel.kt [hunk 5]
```diff
@@ -62,7 +62,7 @@ class OCAuthenticationViewModel(
     fun loginBasic(
         username: String,
         password: String,
-        updateAccountIfAlreadyExists: Boolean = false
+        updateAccountWithUsername: String?
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/authentication/OCAuthenticationViewModel.kt:65]: I would rename this parameter to `updateUserAccountName` for all the classes using it , what do you think?

File: owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/PublicShareDialogFragment.kt (hunk 4 above)
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/PublicShareDialogFragment.kt:614]: BTW i detected a problem:
Second check should not be `shareViaLinkExpirationSwitch?.text` but `shareViaLinkExpirationValue?.text`

File: owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/authentication/OCAuthenticationViewModel.kt (hunk 5 above)
Comment (abelgardep) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/authentication/OCAuthenticationViewModel.kt:65]: We want to update the account that matches with that account name, so i think that the naming is not bad at all.

**Pull request merged**